packages/typescript-to-proptypes/src/injector.ts

```diff
@@ -149,6 +149,13 @@ function plugin(
     ...otherOptions
   } = options;
   const shouldInclude: Exclude<InjectOptions['shouldInclude'], undefined> = (data) => {
+    // key is a reserved prop name in React
+    // e.g. https://github.com/reactjs/rfcs/pull/107
+    // no need to add a prop-type if we won't generate the docs for it.
+    if (data.prop.name === 'key' && data.prop.jsDoc === '@ignore') {
```

Comment:
data.prop.jsDoc === '@ignore' is because we have a custom prop-type for the key in

https://github.com/mui/material-ui/blob/d66f1b691dbd2f8c39004a7a1ebc438cf6e0fdb6/packages/mui-material/src/Snackbar/Snackbar.js#L332-L338

Reply:
`@ignore` is applied when there are no prop descriptions: 

https://github.com/mui/material-ui/blob/d66f1b691dbd2f8c39004a7a1ebc438cf6e0fdb6/scripts/generateProptypes.ts#L209-L213